spring-integration-core/src/main/java/org/springframework/integration/config/ConsumerEndpointFactoryBean.java

```diff
@@ -123,6 +124,7 @@
 
 	private volatile boolean initialized;
 
+	@Reflective // The native image doesn't see this method because its type is not specific
```

Comment:
See the method signature: `setHandler(Object)`.
It works well with regular Java, but in the native image it just doesn't see this method: perhaps it tries to match the setter with the type we provide for the property.
And unfortunately that type is not always just `MessageHandler`. It can be a `ReactiveMessageHandler`.
I cannot have a couple setters because Java Beans introspector cannot guess what setter I'd like to use for XML bean definition parser.